examples/salt_transport/src/example_three_boxes_ZSF.py

```diff
@@ -0,0 +1,386 @@
+        print(self.z.state)
+        print(results)
+        pass
+    elif routine == door_open_on_fresh_side:
+        results = self.z.step_phase_2(t_open_lake, **parameters)
+    elif routine == leveling_to_salt_side:
+        assert t_level > 0
+        print(self.z.state)
+        results = self.z.step_phase_3(t_level, **parameters)
+        print(self.z.state)
+    elif routine == door_open_on_salt_side:
+        print(self.z.state)
+        results = self.z.step_phase_4(t_open_sea, **parameters)
+        print(self.z.state)
+    elif routine in {flushing_2, flushing_4}:
+        results = self.z.step_flush_doors_closed(t_flushing, **parameters)
+    else:
+        raise Exception(f"Unknown routine '{routine}'")
+    print("Salinity lock: {}".format(self.z.state["salinity_lock"]))
+    self.netto_q_in, self.netto_flux = from_zsf_results_to_netto_flux(results, salinity_lake)
+
+    return results
+
+
+def use_zsf_usptream_phase_wise(self, time_step):
+    # netto_q_in = 0.0
+    # netto_flux = 0.0
+    if (
+        int(self.get_current_time()) == int(self.start_lockage_time)
+        and self.locking_index < self.number_of_operations
+    ):
+        print("Now we lock: {}".format(self.start_lockage_time))
+        print("Salinity lock before operation: {}".format(self.z.state["salinity_lock"]))
+
+        parameters = self.lockages[self.locking_index]
+
+        t_open_lake = parameters.pop("t_open_lake")
+        t_open_sea = parameters.pop("t_open_sea")
+        t_level = parameters.pop("t_level")
+        t_flushing = parameters.pop("t_flushing")
+        duration = np.nansum([t_open_lake, t_open_sea, t_level, t_flushing])
+
+        # duration = self.df_time[self.locking_index]
+        self.end_lockage_time = self.start_lockage_time + duration
+        # To ZSF
+
+        salinity_lake = self.get_var(self.active_storage_names[0] + ".C")
+        head_sea_upstream_zsf = self.io.get_timeseries("head_sea_upstream_zsf")[1][int(time_step)]
+        salinity_sea_upstream_zsf = self.io.get_timeseries("salinity_sea_upstream_zsf")[1][
+            int(time_step)
+        ]
+
+        print("Salinity sea: {}".format(salinity_sea_upstream_zsf))
+        print("Salinity lake: {}".format(salinity_lake))
+
+        parameters = self.lockages[self.locking_index]
+
+        parameters["salinity_lake"] = salinity_lake
+        parameters["head_sea"] = head_sea_upstream_zsf
+        parameters["salinity_sea"] = salinity_sea_upstream_zsf
+        parameters["head_lake"] = 10.0
+
+        routine = int(parameters.pop("routine"))
+        """
+        t_open_lake = parameters.pop("t_open_lake")
+        t_open_sea = parameters.pop("t_open_sea")
+        t_level = parameters.pop("t_level")
+        t_flushing = parameters.pop("t_flushing")
+        """
+
+        results = zsf_routines(
+            self, routine, t_level, t_open_lake, t_flushing, t_open_sea, salinity_lake, parameters
+        )
+
+        logger.debug("ZSF results: {}".format(results))
```

Comment:
here you use `logger.debug` but earlier and later you also use `print` I'm not sure which one is correct in this scenario, but in most cases you'll want to use one or the other, so I would double check this